zcash_primitives/src/transaction/builder.rs

```diff
@@ -355,110 +350,63 @@ impl<'a, P: consensus::Parameters, R: RngCore> Builder<'a, P, R> {
         //
         // Signatures -- everything but the signatures must already have been added.
         //
+        let txid_parts = unauthed_tx.digest(TxIdDigester);
 
-        let mut sighash = [0u8; 32];
-        sighash.copy_from_slice(&signature_hash_data(
-            &unauthed_tx,
-            consensus_branch_id,
-            SIGHASH_ALL,
-            SignableInput::Shielded,
-        ));
-
-        #[cfg(feature = "transparent-inputs")]
-        let transparent_sigs = self
-            .transparent_builder
-            .create_signatures(&unauthed_tx, consensus_branch_id);
-
-        let sapling_sigs = self
-            .sapling_builder
-            .create_signatures(prover, &mut ctx, &mut self.rng, &sighash, &tx_metadata)
-            .map_err(Error::SaplingBuild)?;
-
-        #[cfg(feature = "zfuture")]
-        let tze_witnesses = self
-            .tze_builder
-            .create_witnesses(&unauthed_tx)
-            .map_err(Error::TzeBuild)?;
-
-        Ok((
-            Self::apply_signatures(
-                consensus_branch_id,
-                unauthed_tx,
+        let transparent_bundle = unauthed_tx.transparent_bundle.clone().map(|b| {
+            b.apply_signatures(
+                #[cfg(feature = "transparent-inputs")]
```

Comment:
Should this be a separate method for the case with transparent inputs, rather than a method with feature-conditional arguments?

Reply:
That's going to happen in a subsequent PR, when the Orchard builder is integrated.